src/render/tags.rs

```diff
@@ -654,6 +654,43 @@ impl Render for Tag {
             Self::Load => Cow::Borrowed(""),
             Self::SimpleTag(simple_tag) => simple_tag.render(py, template, context)?,
             Self::Url(url) => url.render(py, template, context)?,
+            Self::CsrfToken => match context.get("csrf_token") {
+                Some(token) => {
+                    let bound_token = token.bind(py);
+                    if bound_token.is_truthy().unwrap_or(false) {
+                        let token_str = bound_token.to_string();
+                        if token_str == "NOTPROVIDED" {
+                            Cow::Borrowed("")
+                        } else {
+                            Cow::Owned(format!(
+                                r#"<input type="hidden" name="csrfmiddlewaretoken" value="{}">"#,
+                                html_escape::encode_quoted_attribute(&token_str)
+                            ))
+                        }
+                    } else {
+                        Cow::Borrowed("")
+                    }
+                }
+                None => {
+                    let debug = py
+                        .import("django.conf")
+                        .and_then(|conf| conf.getattr("settings"))
+                        .and_then(|settings| settings.getattr("DEBUG"))
+                        .and_then(|debug| debug.is_truthy())
+                        .unwrap_or(false);
+
+                    if debug {
+                        if let Ok(warnings) = py.import("warnings") {
+                            let _ = warnings.call_method1(
+                                    "warn",
+                                    ("A {% csrf_token %} was used in a template, but the context did not provide the value.  This is usually caused by not using RequestContext.",)
```

Comment:
I added the two spaces between the sentences because even though it doesn't look like there should be two from the warning message in Django's implementation, when testing it via assertion there *were* two spaces. 🤷